tools/shared.py

```diff
@@ -1224,6 +1227,11 @@ def __getattr__(self, attr):
         raise AttributeError
 
     def __setattr__(self, attr, value):
+      for legacy_attr, fixed_values, error_message in self.attrs['REMOVED_SETTINGS']:
+        if attr == legacy_attr and value not in fixed_values:
+          logger.error('Invalid command line option -s ' + attr + '=' + str(value) + ': ' + error_message)
+          sys.exit(1)
```

Comment:
Can you use "exit_with_error()" rather than these two lines?

Reply:
Sure